File: 082_Remove_Duplicates_from_Sorted_List_II/082_3.java
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        if (head == null) return head;

        ListNode pre = null;
        ListNode cur = head;
        while(cur != null && cur.next != null)
        {
            if (cur.val != cur.next.val)
            {
                pre = cur;
                cur = cur.next;
            }
            else
            {
                while(cur.next != null && cur.next.val == cur.val)
                    cur = cur.next;
                if (pre != null)
                    pre.next = cur.next;
                else
                    head = cur.next;
                cur = cur.next;
            }
        }
        return head;
    }
}

File: 082_Remove_Duplicates_from_Sorted_List_II/082_3.py
class Solution:
    def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        if head is None:
            return head

        dummy = ListNode(-1)
        prev, curr = dummy, head
        dummy.next = head
        while curr and curr.next:
            while curr.next and curr.val == curr.next.val:
                curr = curr.next
            if prev.next == curr:
                prev = prev.next
            else:
                prev.next = curr.next
            curr = curr.next
        return dummy.next